rust/private/rustc.bzl

```diff
@@ -645,7 +645,7 @@ def establish_cc_info(ctx, crate_info, toolchain, cc_toolchain, feature_configur
         list: A list containing the CcInfo provider
     """
 
-    if crate_info.is_test or crate_info.type not in ("staticlib", "cdylib") or getattr(ctx.attr, "out_binary", False):
+    if crate_info.is_test or crate_info.type not in ("staticlib", "cdylib", "rlib", "lib") or getattr(ctx.attr, "out_binary", False):
```

Comment:
Could I bother you with updating docs in //rust/private:rust.bzl in `rust_static_library`, `rust_shared_library` and `rust_library`? For first two I'd add  to "This rule provides CcInfo, so it can be used everywhere Bazel expects rules_cc" that the produced library contains core and stdlib, and it uses either system allocator or the one from rust sources.

For rust_library, I'd say that it provides CcInfo, but care needs to be taken to depend on stdlib/allocator correctly.

Reply:
Done - first two in an earlier commit, last one in this commit.